packages/core/src/redocly/registry-api.ts

```diff
@@ -3,22 +3,25 @@ import { RegistryApiTypes } from './registry-api-types';
 const version = require('../../package.json').version;
 
 export class RegistryApi {
-  private readonly baseUrl = `https://api.${process.env.REDOCLY_DOMAIN || 'redoc.ly'}/registry`;
+  private readonly baseUrl = this.getBaseUrl();
 
-  constructor(private accessToken?: string) {}
+  constructor(private accessToken?: string, private domain?: string) {}
 
-  private async request(path = '', options: RequestInit = {}) {
-    if (!this.accessToken) {
+  getBaseUrl() {
+    return `https://api.${this.domain || 'redoc.ly'}/registry`;
+  }
+
+  private async request(path = '', options: RequestInit = {}, accessToken?: string) {
+    if (!this.accessToken && !accessToken) {
       throw new Error('Unauthorized');
     }
 
     const headers = Object.assign({}, options.headers || {}, {
-      authorization: this.accessToken,
+      authorization: accessToken || this.accessToken,
```

Comment:
what is the reason for adding this argument here?

Reply:
refactored